pkg/features/kube_features.go

```diff
@@ -1176,6 +1183,10 @@ var defaultVersionedKubernetesFeatureGates = map[featuregate.Feature]featuregate
 		{Version: version.MustParse("1.34"), Default: true, PreRelease: featuregate.Beta},
 	},
 
+	DeploymentPodReplacementPolicy: {
+		{Version: version.MustParse("1.35"), Default: false, PreRelease: featuregate.Alpha},
+	},
+
 	DeploymentReplicaSetTerminatingReplicas: {
 		{Version: version.MustParse("1.33"), Default: false, PreRelease: featuregate.Alpha},
 	},
```

Comment:
Further below there's `defaultKubernetesFeatureGateDependencies` where we express dependencies between FGs. Since your new feature relies on `DeploymentReplicaSetTerminatingReplicas` it would be good to define it there.

Reply:
That is very cool that we can declare dependencies now. Updated.